Mathlib/Combinatorics/Enumerative/DyckWord.lean

```diff
@@ -77,43 +78,56 @@ instance : AddMonoid DyckWord where
 
 namespace DyckWord
 
-variable (p q : DyckWord)
+variable {p q : DyckWord}
 
 lemma toList_eq_nil : p.toList = [] ↔ p = 0 := by rw [DyckWord.ext_iff]; rfl
+lemma toList_ne_nil : p.toList ≠ [] ↔ p ≠ 0 := toList_eq_nil.ne
+
+@[simp]
+lemma add_eq_zero_iff : p + q = 0 ↔ p = 0 ∧ q = 0 := by
+  simp_rw [← toList_eq_nil]; apply append_eq_nil
+
+lemma add_ne_zero_iff : p + q ≠ 0 ↔ p ≠ 0 ∨ q ≠ 0 := by
+  rw [not_iff_comm, add_eq_zero_iff]; tauto
```

Comment:
Can you generalise this lemma to be just after `add_eq_zero`?

Reply:
See my reply to your previous comment.